ansible/roles/images/tasks/load.yaml

```
@@ -9,12 +9,22 @@
   register: images_cache_find_result
   changed_when: false
 
-- name: import images from files in the local images cache  # noqa no-changed-when
-  shell:
-    executable: /bin/bash
-    cmd: ctr --address {{ containerd_cri_socket }} --namespace k8s.io images import <(gzip --decompress --stdout {{ item }})
+- block:
+    - name: import images from files in the local images cache # noqa no-changed-when
+      shell:
+        executable: /bin/bash
+        cmd: |
+          ctr --address {{ containerd_cri_socket }} --namespace k8s.io images import <(gzip --decompress --stdout {{ item }});
+          if [[ $? -ne 0 ]]; then
+            {{ mindthegap_binary_location_on_remote }} import image-bundle --image-bundle={{ item }};
+          fi
+      register: ctr_images_import
+      failed_when: ctr_images_import.rc != 0
+      changed_when: ctr_images_import.rc == 0
+      with_items: "{{  images_cache_find_result.files | map(attribute='path') | list }}"
+
+    - name: remove mindthegap if present
+      shell:
+        executable: /bin/bash
+        cmd: rm {{ mindthegap_binary_location_on_remote }} || true
```

Should we also move this out of the `block` and always delete it?
```suggestion
    - name: remove mindthegap if present
      file:
        path: "{{ mindthegap_binary_location_on_remote }}"
        state: absent
```

---

It won't be present if we don't set an override with `images_local_bundle_dir`